File: android/app/src/main/java/com/laneshadow/ui/components/atoms/ThemedText.kt
package com.laneshadow.ui.components.atoms

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import com.laneshadow.theme.LocalLaneShadowTheme

/**
 * Themed text variant options
 *
 * - Default: 16sp, normal weight (bodyMedium equivalent)
 * - DefaultSemiBold: 14sp, semibold weight
 */
enum class ThemedTextVariant {
    Default,
    DefaultSemiBold,
}

/**
 * ThemedText atom component
 *
 * Lightweight themed text wrapper that applies LaneShadow theme tokens.
 * Use this component for consistent typography throughout the app.
 *
 * @param text The text content to display
 * @param variant Typography variant (Default or DefaultSemiBold)
 * @param color Optional color override (defaults to onSurface)
 * @param modifier Modifier for the text
 * @param testID Optional test identifier for UI testing
 */
@Composable
fun ThemedText(
    text: String,
    variant: ThemedTextVariant = ThemedTextVariant.Default,
    color: Color? = null,
    modifier: Modifier = Modifier,
    testID: String? = null,
) {
    val theme = LocalLaneShadowTheme.current

    // Default color from theme
    val textColor = color ?: theme.colors.onSurface.default

    // Typography style based on variant
    val textStyle = when (variant) {
        ThemedTextVariant.Default -> androidx.compose.ui.text.TextStyle(
            fontSize = theme.type.body.md.fontSize,
            fontWeight = theme.type.body.md.fontWeight,
            letterSpacing = 0.sp,
        )
        ThemedTextVariant.DefaultSemiBold -> androidx.compose.ui.text.TextStyle(
            fontSize = 14.sp,
            fontWeight = FontWeight.SemiBold,
            letterSpacing = 0.sp,
        )
    }

    Text(
        text = text,
        style = textStyle,
        color = textColor,
        modifier = modifier,
    )
}

File: ios/LaneShadow/Views/Atoms/ThemedText.swift
import LaneShadowTheme
import SwiftUI

// MARK: - Themed Text Variant Enum

/**
 * Themed text typography variants
 *
 * Following RN wrapper API from react-native/components/ui/themed-text.tsx
 * Matrix reference: .spec/prds/native-rewrite/matrices/ui/atoms/ThemedText.md
 */
public enum LSThemedTextVariant {
    case defaultSize // bodyMedium: size 16, regular weight
    case defaultSemiBold // size 14, semibold weight
}

// MARK: - Themed Text Component

/**
 * Themed text component
 *
 * Following RN wrapper API from react-native/components/ui/themed-text.tsx
 * Matrix reference: .spec/prds/native-rewrite/matrices/ui/atoms/ThemedText.md
 *
 * ## Design Tokens Used
 * - Colors: `theme.colors.onSurface.default` (default text color)
 * - Typography:
 *   - Default: 16pt regular (bodyMedium)
 *   - SemiBold: 14pt semibold
 *
 * ## Parameters
 * - Parameters:
 *   - text: Text content to display
 *   - variant: Typography variant (default, defaultSemiBold)
 *   - color: Optional color override (defaults to onSurface)
 *   - testID: Test identifier for UI testing
 */
public struct LSThemedText: View {
    @Environment(\.theme) private var theme

    private let text: String
    private let variant: LSThemedTextVariant
    private let color: Color?
    private let testID: String?

    public init(
        _ text: String,
        variant: LSThemedTextVariant = .defaultSize,
        color: Color? = nil,
        testID: String? = nil
    ) {
        self.text = text
        self.variant = variant
        self.color = color
        self.testID = testID
    }

    // MARK: - Typography Computed Properties

    private var font: Font {
        switch variant {
        case .defaultSize:
            .system(size: 16, weight: .regular)
        case .defaultSemiBold:
            .system(size: 14, weight: .semibold)
        }
    }

    private var textColor: Color {
        color ?? theme.colors.onSurface.default
    }

    // MARK: - Body

    public var body: some View {
        Text(text)
            .font(font)
            .foregroundStyle(textColor)
            .accessibilityIdentifier(testID ?? "themed-text")
    }
}
